app/code/Magento/CatalogUrlRewrite/Model/Product/Attribute/Backend/UrlKey.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Magento\CatalogUrlRewrite\Model\Product\Attribute\Backend;
+
+use Magento\Catalog\Model\Product;
+use Magento\CatalogUrlRewrite\Model\ResourceModel\Product\IsUniqueUrlKey;
+
+class UrlKey extends \Magento\Eav\Model\Entity\Attribute\Backend\AbstractBackend
+{
+    /**
+     * @var IsUniqueUrlKey
+     */
+    private $urlKey;
+
+    /**
+     * @param IsUniqueUrlKey $urlKey
+     */
+    public function __construct(
+        IsUniqueUrlKey $urlKey
+    ) {
+        $this->urlKey = $urlKey;
+    }
+
+    /**
+     * @param  Product $object
+     * @return bool
+     * @throws \Magento\Framework\Exception\LocalizedException
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function validate($object)
```

Comment:
Hi @riccardougolini,
Also missing argument typing and return typing

Reply:
hi @williankeller, unfortunately I can't type the method more than this cause on \Magento\Eav\Model\Entity\Attribute\Backend\AbstractBackend the method validate does not have a complete typing.